blockly-core/core/utils/xml.js

```diff
@@ -17,6 +17,12 @@
  * limitations under the License.
  */
 
+/* global Blockly, goog */
+
+// Ignore "Expected a conditional expression and instead saw an
+// assignment" warnings.
+/* jshint -W084 */
```

Comment:
If true that blockly-core is only linted by pre-commit hook, it seems that it should either (a) not be linted by precommit hook or (b) be linted by precommit hook and by some other part of our process. b is obviously better, but might require some amount of work to clean things up.